libs/native/src/test21/java/org/elasticsearch/nativeaccess/jdk/vec/NativeSimilarityTests.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.nativeaccess.jdk.vec;
+
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+
+import org.elasticsearch.nativeaccess.VectorSimilarityFunctionsTests;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+import java.lang.foreign.Arena;
+import java.lang.foreign.MemorySegment;
+import java.util.stream.IntStream;
+
+import static org.hamcrest.Matchers.containsString;
+
+public class NativeSimilarityTests extends VectorSimilarityFunctionsTests {
```

Comment:
> This can move back to the o.e.na.jdk package? 

Ah, yes. Moved and renamed.

VectorSimilarityFunctionTests has some basic checks on the public interface of retrieving a VectorSimilarityFunction instance - optional is present or empty. These checks, while JDK-version specific, can be run on all platforms and JDKs.

NativeSimilarityTests is JDK 21 specific - since it uses MemorySegment directly, and asserts the results of computations.